documentation/p1950.md

```diff
@@ -622,39 +628,49 @@ The template parameter `T` of `indirect_value` may be an incomplete type.
 ### X.Z.6 Class template `indirect_value` modifiers [indirect_value.modifiers]
 <code>void swap(indirect_value<T>& p) noexcept;</code>
 * Effects: Exchanges the contents of `p` and `*this`.
-* Remarks: The method shall work with incomplete pointer type for `T`.
+* Remarks: The function shall work with incomplete pointer type for `T`.
 
 ### X.Z.7 Class template `indirect_value` observers [indirect_value.observers]
 ```
 T& operator*();
 const T& operator*() const;
 ```
-* Requires: `bool(*this)`.
+* Preconditions: `bool(*this)` is true.
 * Returns: A reference to the owned object.
 
 ```
 T* operator->() noexcept;
 const T* operator->() const noexcept;
 ```
-* Requires: `bool(*this)`.
+* Preconditions: `bool(*this)` is true.
 * Returns: A pointer to the owned object.
-* Remarks: The method shall work with incomplete pointer type for `T`.
+* Remarks: The function shall work with incomplete pointer type for `T`.
 
 <code>explicit operator bool() const noexcept;</code>
 * Returns: false if the `indirect_value` is empty, otherwise true.
-* Remarks: The method shall work with incomplete pointer type for `T`.
+* Remarks: The function shall work with incomplete pointer type for `T`.
 
 ### X.Z.8 Class template `indirect_value` creation [indirect_value.creation]
 
 ```cpp
-template <class T, class U=T, class ...Ts> indirect_value<T>
-  make_indirect_value(Ts&& ...ts);
+template <class T, class U=T, class ...Ts>
+indirect_value<T> make_indirect_value(Ts&& ...ts);
+```
+
+* Constraints: `is_constructible_v<U, Ts...>` is true.
+* Preconditions: `U` meets the `Cpp17CopyConstructible` requirements.
+* Returns: A `indirect_value<T>` owning an object of type direct-non-list-initialized with `std::forward<Ts>(ts)...`.
+* Preconditions: `is_same_v<C, default_copy> && is_same_v<D, default_delete>`.
+
+```cpp
+template <class T, class U = T, class A = std::allocator<U>, class... Ts>
+indirect_value<T> make_indirect_value(std::allocator_arg_t, A& a, Ts&&... ts)  
 ```
 
 * Constraints: `is_constructible_v<U, Ts...>` is true.
-* Expects: `U` meets the `Cpp17CopyConstructible` requirements.
+* Preconditions: `U` meets the `Cpp17CopyConstructible` requirements.
 * Returns: A `indirect_value<T>` owning an object of type direct-non-list-initialized with `std::forward<Ts>(ts)...`.
-* Requires `is_same_v<C, default_copy> && is_same_v<D, default_delete>`.
+* Preconditions `is_same_v<C, allocator_copy> && is_same_v<D, allocator_delete>`.
```

Comment:
This requires clarification as types `allocator_copy`  and `allocator_delete` are not yet introduced, but are in the references implementation in the detail namespace:
- https://github.com/jbcoe/indirect_value/blob/a74a6d70569cf9853a7af22dde15fae0f6fccd70/indirect_value.h#L119
- https://github.com/jbcoe/indirect_value/blob/a74a6d70569cf9853a7af22dde15fae0f6fccd70/indirect_value.h#L128

Ideally these would be an implementation detail.  Do we need to define these in the spec?

Reply:
I'd keep them as implementation details if possible.